source/application/BioXAS/BioXASAppController.cpp

```diff
@@ -459,6 +460,12 @@ QWidget* BioXASAppController::createComponentView(QObject *component)
 			componentView = new AMMotorGroupView(cryostatStage->motorGroup(), AMMotorGroupView::CompactView);
 			componentFound = true;
 		}
+
+		BioXASFilterFlipper *filterFlipper = qobject_cast<BioXASFilterFlipper*>(component);
+		if (!componentFound && filterFlipper) {
+			componentView = new BioXASFilterFlipperView(filterFlipper);
+			componentFound = true;
+		}
```

Comment:
I've never used a factory pattern before, I'll have to look into it. It would be useful in this case?